File: frontend/src/types/api/review.ts
export interface Review {
  id: string;
  organizationId: string;
  membershipId: string;
  residentName: string;
  rating: number;
  body: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateReviewInput {
  rating: number;
  body: string;
}

export interface UpdateReviewInput {
  rating: number;
  body: string;
}

File: backend/src/main/java/com/dwellio/domain/entity/Review.java
package com.dwellio.domain.entity;

import com.dwellio.common.entity.SoftDeletableEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.util.UUID;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "reviews")
@Getter
@Setter
public class Review extends SoftDeletableEntity {

    @Id
    @Column(columnDefinition = "uuid")
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "organization_id", nullable = false)
    private Organization organization;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "membership_id", nullable = false, unique = true)
    private Membership membership;

    @Column(nullable = false)
    private short rating;

    @Column(columnDefinition = "TEXT")
    private String body;
}
